src/org/labkey/test/components/ui/permissions/PermissionsRowBase.java

```diff
@@ -150,17 +151,31 @@ protected PermissionsMember findGroup(String name)
 
 
     @Override
-    protected ElementCache newElementCache()
-    {
-        return new ElementCache();
-    }
+    protected abstract ElementCache newElementCache();
 
     protected class ElementCache extends WebDriverComponent<?>.ElementCache
     {
         final WebElement rowContainer = Locator.byClass("container-expandable-grey").findWhenNeeded(this);
 
-        final ReactSelect memberSelect = ReactSelect.finder(getDriver()).findWhenNeeded(this)
-                .setOptionLocator(ReactSelect.Locators.option::endsWith);
+        private final ReactSelect groupMemberSelect;
+        private final ReactSelect userMemberSelect;
+
+        protected ElementCache(Function<String, Locator> groupOptionLocator, Function<String, Locator> userOptionLocator)
+        {
+            // Use same WebElement for both selects but with different option locators.
+            ReactSelect rawMemberSelect = ReactSelect.finder(getDriver()).findWhenNeeded(this);
+
+            groupMemberSelect = new ReactSelect(rawMemberSelect);
```

Comment:
The nice thing about centralizing/sharing the configuration of how these options will be matched in the select is that it discovers that the product isn't consistent about how it shows these options
The downside to doing it is that it doesn't work with the product as is; the assumption that the product will present role options consistently from GroupRow to RoleRow turns out to be wrong.